app/src/server/api/routers/kage.ts

```diff
@@ -537,21 +548,61 @@ export const kageRouter = createTRPCRouter({
       if (nPendingRequests > 0) {
         return errorResponse("Cannot toggle while there are pending challenges");
       }
-      const secondsSinceOpen = secondsPassed(userVillage.openForChallengesAt);
-      if (secondsSinceOpen < KAGE_CHALLENGE_OPEN_FOR_SECONDS) {
-        return errorResponse(
-          `Please wait ${Math.floor(KAGE_CHALLENGE_OPEN_FOR_SECONDS - secondsSinceOpen)} seconds before toggling`,
-        );
-      }
 
-      // Update
-      await ctx.drizzle
-        .update(village)
-        .set({
-          openForChallenges: !userVillage.openForChallenges,
-          openForChallengesAt: new Date(),
+      // Check cooldown from last toggle (using actionLog)
+      const lastToggle = await ctx.drizzle
```

Comment:
This can be fetched together with user, requests, userVillage. Cost-wise it's better to run the queries as much in parallel as possible (as long as they're not crazy heavy), since it improved endpoint speed due to fewer round-abouts to DB, and also lowers the CPU time of the lambda function.